src/cirq_iqm/iqm_sampler.py

```diff
@@ -164,4 +187,43 @@ def _send_circuits(
         if results.measurements is None:
             raise RuntimeError('No measurements returned from IQM quantum computer.')
 
-        return [{k: np.array(v) for k, v in measurements.items()} for measurements in results.measurements]
+        return (
+            [{k: np.array(v) for k, v in measurements.items()} for measurements in results.measurements],
+            job_id,
+            results.metadata.calibration_set_id,
+            results.metadata.request,
+        )
+
+    def _create_iqm_result(
+        self,
+        result_dict: cirq.study.ResultDict,
+        job_id: UUID,
+        calibration_set_id: UUID,
+        request: RunRequest,
+    ) -> IQMResult:
+        """
+        Creates an IQMResult instance with the given attributes.
+        """
+        return IQMResult(
+            result_dict=result_dict,
+            job_id=job_id,
+            calibration_set_id=calibration_set_id,
+            request=request,
+        )
+
+
+@dataclass
+class IQMResult:
```

Comment:
We want to stick with the cirq function signatures. To achieve what you want I would suggest to have `IQMResult` extend `cirq.Result` and add a `metadata` field, which then can hold the information that you added here.

Reply:
Thanks Matthias! Will work on this soon!

Reply:
Hi @q-mat-beu I've done some work based on your comments:
- Instead of extended `cirq.Result` I chose to extend `cirq.ResultDict`. This is because `cirq.Result` is an abstract class and `cirq.ResultDict` contains all of the functionality we wish `IQMResult` to hold. 
- `IQMResult` now has an attribute `metadata` using the `ResultMetadata` dataclass. 
- `run_sweep`, `run_iqm_batch` and `_send_circuits` have been updated for the new class. 

The coverage of `iqm_sampler` is now 89%. I will add some unit tests to try and improve this.